metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorTableService.java

```diff
@@ -423,6 +428,23 @@ public void rename(
                 "Renaming tables is disabled in catalog " + catalogName, null);
         }
         try {
+            if (onRenameConvertToExternal) {
+                //
+                // If this is a managed table(EXTERNAL=FALSE), then convert it to an external table before renaming it.
+                // We do not want the metastore to move the location/data.
+                //
+                final Table table = metacatHiveClient.getTableByName(oldName.getDatabaseName(), oldName.getTableName());
+                Map<String, String> parameters = table.getParameters();
+                if (parameters == null) {
+                    parameters = Maps.newHashMap();
+                    table.setParameters(parameters);
+                }
+                if (!parameters.containsKey(PARAMETER_EXTERNAL)
+                    || parameters.get(PARAMETER_EXTERNAL).equalsIgnoreCase("FALSE")) {
+                    parameters.put(PARAMETER_EXTERNAL, "TRUE");
```

Comment:
You may want to log the conversion here.

Reply:
Keeping it up describes the logic for the whole if block.